File: leetcode/128_longest_consecutive_sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        a = set(nums)
        res = 0
        for e in nums:
            if e-1 not in a:
                c = 1
                num = e + 1
                while num in a:
                    num += 1
                    c += 1
                res = max(res, c)
        return res

File: leetcode/128_longest_consecutive_sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        int result = 0;
        for(int& num : nums) {
            int c = 1;
            int curr = num;
            while(s.find(--curr) != s.end()) {
                c++;
                s.erase(curr);
            }
            curr = num;
            while(s.find(++curr) != s.end()) {
                c++;
                s.erase(curr);
            }
            result = max(result, c);
        }
        return result;
    }
};
